### PR TITLE
Compile `test_remote_optimizer.cpp`

### PR DESCRIPTION
**Pull request by Y--**
Also update amalgamation list

### CODE REVIEW

File: test/extension/loadable_extension_optimizer_demo.cpp [hunk 1]
```diff
@@ -21,6 +22,46 @@ using namespace duckdb;
 #include <sys/types.h>
 #include <arpa/inet.h>
 
+namespace duckdb {
+
+//! LogicalChunkGet represents a scan operation from a ChunkCollection
+class LogicalChunkGet : public LogicalOperator {
```
Comment (Mytherin) [inline test/extension/loadable_extension_optimizer_demo.cpp:28]: Could we perhaps rewrite the test to use a `ColumnDataCollection` instead?

File: test/extension/CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+
 build_loadable_extension_directory(
   loadable_extension_demo test/extension loadable_extension_demo.cpp
   ${IGNORE_WARNINGS})
+
+build_loadable_extension_directory(
+  loadable_extension_optimizer_demo test/extension ../extension/loadable_extension_optimizer_demo.cpp
+  ${IGNORE_WARNINGS})
+
+set(TEST_EXT_OBJECTS
```
Comment (Mytherin) [inline test/extension/CMakeLists.txt:13]: This should likely only be built on certain OS configurations, no?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! I think it's a good idea to build this test, but some more work is required to get this through the CI (hence also why I skipped it in the initial serialization PR :)). Some comments:

File: test/extension/loadable_extension_optimizer_demo.cpp (hunk 1 above)
Comment (Y--) [reply test/extension/loadable_extension_optimizer_demo.cpp]: Thanks! Works like a charm. I've updated it, let me know what you think

File: src/include/duckdb/common/types/column_data_collection.hpp [hunk 3]
```diff
@@ -135,9 +135,9 @@ class ColumnDataCollection {
 	void ScanAtIndex(ColumnDataParallelScanState &state, ColumnDataLocalScanState &lstate, DataChunk &result,
 	                 idx_t chunk_index, idx_t segment_index, idx_t row_index) const;
 
-private:
 	//! Initialize the column data collection
 	void Initialize(vector<LogicalType> types);
+private:
```
Comment (Y--) [inline src/include/duckdb/common/types/column_data_collection.hpp:140]: @Mytherin I had to expose the `Initialize` method to get the test working. Let me know if you prefer me to keep it private and initialize it some other way (I would have to expose the `copy_functions` collection though)

**Pull request closed without merging**